test/encode.spec.ts

```diff
@@ -80,4 +82,61 @@ describe('encode', () => {
       expect(output[i]).to.be.an.instanceOf(Uint8Array)
     }
   })
+
+  it('should not encode message data that is too long', async () => {
+    const input = [new Uint8Array(MAX_DATA_LENGTH + 1)] // Create a buffer that exceeds the max length
+    try {
+      await pipe(
+        input,
+        (source) => lp.encode(source),
+        async (source) => all(source)
+      )
+      throw new Error('Expected error not thrown') // Fail the test if the error isn't thrown
+    } catch (err: any) {
+      expect(err).to.be.an.instanceof(InvalidDataLengthError)
+      expect(err.code).to.equal('ERR_MSG_DATA_TOO_LONG') // Check the error code
+    }
```

Comment:
A small nit but you can do this sort of thing without resorting to exceptions for flow control with:

```suggestion
    await expect(pipe(...)).to.eventually.be.rejected
      .with.property('code', 'ERR_MSG_DATA_TOO_LONG')
```

Reply:
ok thanks